jupyterlab/handlers/announcements.py

```diff
@@ -0,0 +1,258 @@
+"""Announcements handler for JupyterLab."""
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import abc
+import hashlib
+import json
+import xml.etree.ElementTree as ET
+from dataclasses import asdict, dataclass, field
+from datetime import datetime
+from typing import Awaitable, Optional
+
+from jupyter_server.base.handlers import APIHandler
+from jupyterlab_server.translation_utils import translator
+from packaging.version import parse
+from tornado import httpclient, web
+
+from .._version import __version__
+
+ISO8601_FORMAT = "%Y-%m-%dT%H:%M:%S%z"
+JUPYTERLAB_LAST_RELEASE_URL = "https://pypi.org/pypi/jupyterlab/json"
+JUPYTERLAB_RELEASE_URL = "https://github.com/jupyterlab/jupyterlab/releases/tag/v"
+
+
+@dataclass(frozen=True)
+class Notification:
+    """Notification
+
+    Attributes:
+        message: Notification message
+        type: Notification type — ["default", "error", "info", "success", "warning"]
+        options: Notification options
+    """
+
+    createdAt: float
+    message: str
+    modifiedAt: float
+    type: str = "default"
+    options: dict = field(default_factory=dict)
+
+
+class CheckForUpdateABC(abc.ABC):
+    """Abstract class to check for update.
+
+    Args:
+        version: Current JupyterLab version
+
+    Attributes:
+        version - str: Current JupyterLab version
+        logger - logging.Logger: Server logger
+    """
+
+    def __init__(self, version: str) -> None:
+        self.version = version
+
+    @abc.abstractmethod
+    async def __call__(self) -> Awaitable[Optional[str]]:
+        """Get the notification message if a new version is available.
+
+        Returns:
+            The notification message or None if there is not update.
+        """
+        raise NotImplementedError("CheckForUpdateABC.__call__ is not implemented")
+
+
+class CheckForUpdate(CheckForUpdateABC):
+    """Default class to check for update.
+
+    Args:
+        version: Current JupyterLab version
+
+    Attributes:
+        version - str: Current JupyterLab version
+        logger - logging.Logger: Server logger
+    """
+
+    async def __call__(self) -> Awaitable[Optional[str]]:
+        """Get the notification message if a new version is available.
+
+        Returns:
+            The notification message or None if there is not update.
+        """
+        http_client = httpclient.AsyncHTTPClient()
+        try:
+            response = await http_client.fetch(
+                JUPYTERLAB_LAST_RELEASE_URL,
+                headers={"Content-Type": "application/json"},
+            )
+            data = json.loads(response.body).get("info")
+            last_version = data["version"]
```

Comment:
I was wondering if this would pick up pre-releases. Here the `version` corresponds to a stable version, which is good :+1: 

![image](https://user-images.githubusercontent.com/591645/202394889-4d738fdf-3d60-429e-9e35-e74e4bc1b18e.png)

Reply:
Yes this reflect the default page a user will see when going to PyPI.org (aka ignoring the pre releases).